### DISCUSSION TITLE
query regarding Vscode extension for premake

### DISCUSSION THREAD

**Discussion by holychicken99** [premake/premake-core#1766] [Ideas]
Hi Premake team ,
I have built and deployed a premake-snippet extension to [Vs code marketplace.](https://marketplace.visualstudio.com/items?itemName=holychicken99.premake-snippets) Is it alright to use the premake logo for its icon ?.
also is it possible that this extension be displayed in the [premake website](https://premake.github.io/community/modules) as an IDE module for VScode. 
Thank you

**Comment by Ethan-Bit-256** [comment]
As a visual studio code user, I think that sounds awesome! What's it called?

**Reply by KyrietS** [reply]
It's called [premake-snippets](https://marketplace.visualstudio.com/items?itemName=holychicken99.premake-snippets).

**Comment by KyrietS** [comment]
> Is it alright to use the premake logo for its icon ?

I don't feel fully empowered to answer this, but I personally don't see any objections.

> is it possible that this extension be displayed in the premake website as an IDE module for VScode.

Your extension is not a module, so it should not be placed in Community/Modules. An IDE Module would be something that generates a .vscode project with a preconfigured task.json to build, compile, etc. And it's usage would be: `premake5 vscode` (for example).

But we are open to suggestions. Maybe you should make a new section Community/Extensions? Feel free to do a PR to our website 🤠

**Comment by starkos** [comment]
> Is it alright to use the premake logo for its icon ?

Yes, I have no objection to this.

>  is it possible that this extension be displayed in the premake website as an IDE module for VScode

Also yes, though as @KyrietS mentioned this isn't actually a Premake module so you'll need to create a different section—feel free to submit a PR with the website changes and we can discuss there.
